### PR TITLE
Biowaste code refactor

### PR DESCRIPTION
**Pull request by DanRStevens**
Suggestions from #753

### CODE REVIEW

File: OPHD/Things/Structures/Recycling.h [hunk 1]
```diff
@@ -38,7 +38,7 @@ class Recycling : public Structure
 	 */
 	virtual int residentialSupportCount() const
 	{
-	return ResidentialSupportCount;
+		return ResidentialSupportCount;
```
Comment (ldicker83) [inline OPHD/Things/Structures/Recycling.h:41]: Yeah, I saw this and wanted to kick myself. Whoops. -_-

File: OPHD/States/MapViewStateTurn.cpp [hunk 2]
```diff
@@ -351,28 +351,18 @@ void MapViewState::updateBiowasteRecycling()
...
-		
 		if (!recyclingFacility->operational()) { continue; } // Consider a different control structure
 
 		Recycling* recycling = static_cast<Recycling*>(recyclingFacility);
-
-		for (; residenceIterator != residences.end(); ++residenceIterator)
+		for (int count = 0; count < recycling->residentialSupportCount(); ++count)
 		{
+			if (residenceIterator == residences.end()) {
+				return; // No more residences, so don't waste time iterating over remaining recycling facilities
```
Comment (ldicker83) [inline OPHD/States/MapViewStateTurn.cpp:360]: Should either move the brace placement for consistency or make this a single line with the comment above it.

**Comment by ldicker83** [comment]
So aside from the formatting thing I noted to fix, this looks pretty good.

Good example of how I can over complicate code by being too tired :D

File: OPHD/States/MapViewStateTurn.cpp (hunk 2 above)
Comment (DanRStevens) [reply OPHD/States/MapViewStateTurn.cpp:360]: Oh right, I meant to match up the brace style. Guess I got distracted on that one.

**Pull request merged**